core/src/io/github/eng12020team24/project1/characters/Auber.java

```diff
@@ -59,8 +61,12 @@ public void move(float deltaTime) {
         }
         // rotation = 180 is true when moving up.  But when moving up, rotation should be 90 (sin 1, cos 0)
         if (Gdx.input.isKeyPressed(Keys.W) || Gdx.input.isKeyPressed(Keys.A) || Gdx.input.isKeyPressed(Keys.S) || Gdx.input.isKeyPressed(Keys.D)) {
-            xPos += Math.round(deltaTime * 4 * 32 * Math.cos(Math.toRadians(rotation - 90)));
-            yPos += Math.round(deltaTime * 4 * 32 * Math.sin(Math.toRadians(rotation - 90)));
+            float newX = xPos + Math.round(deltaTime * 4 * 32 * Math.cos(Math.toRadians(rotation - 90)));
+            float newY = yPos + Math.round(deltaTime * 4 * 32 * Math.sin(Math.toRadians(rotation - 90)));
+            if (!map.doesRectCollideWithMap(newX-16,newY-16,30,30)) {
```

Comment:
I'm not entirely sure on this if statement as it doesnt let you slide along the wall.  That said, I can't think of a better way of doing this.